plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1074,19 +1070,15 @@ public void setTableComment(ConnectorSession session, ConnectorTableHandle table
         try {
             long commitVersion = handle.getReadVersion() + 1;
 
-            List<String> partitionColumns = getPartitionedBy(tableMetadata.getProperties());
-            List<DeltaLakeColumnHandle> columns = tableMetadata.getColumns().stream()
-                    .filter(column -> !column.isHidden())
-                    .map(column -> toColumnHandle(column, column.getName(), column.getType(), partitionColumns))
-                    .collect(toImmutableList());
+            List<DeltaLakeColumnHandle> columns = getColumnHandles(tableMetadata, column -> !column.isHidden());
```

Comment:
in `setTableComment` we shouldn't need to convert delta schema to connectortablemetadata to deltalakecolumnhandles back to delta schema.
this process may be lossy

(no change requested now)

Reply:
Follow up PR https://github.com/trinodb/trino/pull/17176